src/kayenta/stages/kayentaStage/kayentaStage.transformer.ts

```diff
@@ -26,10 +30,44 @@ export class KayentaStageTransformer implements ITransformer {
 
         // For now, a 'kayentaCanary' stage should only have an 'aggregateCanaryResults' task, which should definitely go last.
         stage.tasks = [...syntheticCanaryStages, ...stage.tasks];
+      } else if (stage.type === CREATE_SERVER_GROUP) {
+        OrchestratedItemTransformer.defineProperties(stage);
+        const locations = stage.context['deploy.server.groups'] && Object.keys(stage.context['deploy.server.groups']);
+        stage.name = `Deploy ${stage.context.freeFormDetails}${locations ? ' in ' + locations.join(', ') : ''}`;
+        stage.parentStageId = kayentaStage.id;
       }
     });
 
-    execution.stages = execution.stages.filter(stage => !stagesToRenderAsTasks.includes(stage));
+    const deployCanaryServerGroupsStage = execution.stages.find(
+      ({ parentStageId, type }) => parentStageId === kayentaStage.id && type === DEPLOY_CANARY_SERVER_GROUPS,
+    );
+
+    if (deployCanaryServerGroupsStage && (deployCanaryServerGroupsStage.outputs.deployedServerGroups || []).length) {
+      const [{ controlScope, experimentScope }] = deployCanaryServerGroupsStage.outputs.deployedServerGroups;
+      kayentaStage.outputs = { ...kayentaStage.outputs, controlScope, experimentScope };
+    }
```

Comment:
When an automatic deploy happens, the only place to get the actual scopes that were used is the output of the `deployCanaryServerGroups` stage that wraps the whole deployment. Thing is, we want to hide that stage from the controller when it succeeds.

Since we end up using these scopes in the `kayentaCanary` stage anyway, I opted to just inject the scopes into that stage's outputs.